<launch>
    <!-- 参数定义 -->
    <arg name="markerIds" default="100,101,102,103,104"/> <!-- 多个markerID，用逗号分隔 -->
    <arg name="markerSize" default="0.1"/> <!-- 单位：米 -->
    <arg name="camera_info_topic" default="/camera/color/camera_info"/>
    <arg name="image_topic" default="/camera/color/image_raw"/>
    <arg name="depth_topic" default="/camera/depth/image_raw"/>
    <arg name="marker_frame_prefix" default="aruco_marker_frame"/>
    <arg name="reference_frame" default="camera_link"/>
    <arg name="camera_frame" default="camera_color_optical_frame"/>
    <arg name="corner_refinement" default="LINES"/> <!-- NONE, HARRIS, LINES, SUBPIX -->

    <!-- 启动节点 -->
    <node pkg="aruco_ros" type="single" name="aruco_single" output="screen">
        <!-- 主题映射 -->
        <remap from="/camera_info" to="$(arg camera_info_topic)"/>
        <remap from="/image" to="$(arg image_topic)"/>
        <remap from="/camera/depth/image_raw" to="$(arg depth_topic)"/>

        <!-- 参数设置 -->
        <param name="image_is_rectified" value="true"/>
        <param name="marker_size" value="$(arg markerSize)"/>
        <param name="marker_ids" value="$(arg markerIds)"/> <!-- 传递多个markerID -->
        <param name="reference_frame" value="$(arg reference_frame)"/>
        <param name="camera_frame" value="$(arg camera_frame)"/>
        <param name="marker_frame_prefix" value="$(arg marker_frame_prefix)"/>
        <param name="corner_refinement" value="$(arg corner_refinement)"/>
    </node>
</launch>